src/Utils.js

```diff
@@ -113,6 +114,13 @@ function childrenOfNode(node) {
   return childrenToArray(children);
 }
 
+function isTextualNode(node) {
+  return typeof node === 'string' || typeof node === 'number';
+}
+
+export function isReactElementAlike(arg) {
+  return React.isValidElement(arg) || isTextualNode(arg) || Array.isArray(arg);
```

Comment:
Well, it is for sure a common issue for people that they assume selectors are allowed here. 

Maybe we should allow them and try to merge those 2 different logics of selectors and nodeEquals into one? Although I ain't sure if that's even possible elegantly.